dev/snippets/lib/snippets.dart

```diff
@@ -227,6 +227,7 @@ class SnippetGenerator {
       if (match != null) { // If we saw the start or end of a code block
         inCodeBlock = !inCodeBlock;
         if (match.namedGroup('language') != null) {
+          // ignore: unnecessary_null_checks
```

Comment:
I think this means the lint is not ready yet for our code base. We don't want these ignores everywhere.

Reply:
So should I comment in lint?

Reply:
Actually(after checking) the lint already handle self promotion to non-null (`nullable = nullable!` is ok) and I think it would defeat the lint to allow any `nullable = x!`.

A workaround to avoid `// ignore` would be here:

```dart
language = match[1];
language = language!;
```

As there's only 2 cases where this happens I tend to see this workaround as _acceptable_.